### ISSUE TITLE
pwg:7452

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/csl-orig#1173]
date:	04/26/2023 07:06:50
dict:	pwg
Lnum:	7452
hw:	asūy
old:	a̱sū̱yannabhyàcākaśa̱ṃ tasmā̀ aspṛhaya̱ṃ punàḥ
new:	a̱sū̱yannabhyàcākaśa̱ṃ tasmā́ aspṛhaya̱ṃ punáḥ
comm:	Typo LB (It is one of the examples with an accent typo. Other words also have to be corrected.)

**Comment by funderburkjim** [comment]
<img width="644" alt="image" src="https://github.com/sanskrit-lexicon/csl-orig/assets/6393033/852433b6-043d-483b-8f87-e8810da3a6b0">

As coded in pwg.txt (with slp1 transliteration)
```
{#a\sU\yannaBya^cAkaSa\M tasmA^ aspfhaya\M puna^H#}
```
display Devanagari:

display IAST:
a̱sū̱yannabhyàcākaśa̱ṃ tasmā̀ aspṛhaya̱ṃ punàḥ

display Devanagari:
अ॒सू॒यन्नभ्य॑चाकशं॒ तस्मा॑ अस्पृहयं॒ पुनः॑

**Comment by funderburkjim** [comment]
The user suggests changing accents on two words from svarita to udAtta (have I got the names right?): 
  tasmā̀  -> tasmā́   and   punàḥ -> punáḥ

But udAtta accent in Devanagari uses the special 'superscript u' in PWG
e.g. in previous word:

<img width="501" alt="image" src="https://github.com/sanskrit-lexicon/csl-orig/assets/6393033/912ef230-8d21-40c2-bb1b-9a45e1d75cce">

## Conclusion
No change required.  The user's two changes are wrong, as the accent is not
udAtta but svarita.

**Comment by funderburkjim** [comment]
@Andhrabharati Do you agree with my analysis of no change?

**Comment by Andhrabharati** [comment]
Yes, you are right.

**Comment by funderburkjim** [comment]
The user (Yuzuki T.)  provided additional feedback:
> I still believe there are errors in the accents. 
  In PWG, the special symbols for udātta should only be applied to the headwords, while the examples should use regular accent marks. 
Initially, I suggested a faulty proposal by simply replacing "svarita" with "udātta," 
but I believe a more accurate transliteration would be 
"a̱sū̱yánnabhyàcākaśa̱ṃ tásmā aspṛhaya̱ṃ púnaḥ" 
 (see https://vedaweb.uni-koeln.de/rigveda/view/id/10.135.2). 
Please review the accent rules again (Vedic Grammar for Students pp.448ff https://archive.org/details/vedicgrammarfor00macduoft/page/448/mode/2up).

I am not competent to evaluate this suggestion.  Maybe others (@drdhaval2785 , @Andhrabharati , @SergeA , @gasyoun or others) may wish to comment.  

Is there a larger issue here with regard to accent representation in the CDSL version of pwg?

**Comment by funderburkjim** [comment]
Also, perhaps @fxru or @maltenth may have opinions or insights into pwg accents in cdsl version.

**Comment by Andhrabharati** [comment]
We may note that Vedaweb has resorted to the ISO transliteration (which has a wide and full coverage of all Indic scripts), and not to IAST (which has some limitations)

**Comment by Andhrabharati** [comment]
> Is there a larger issue here with regard to accent representation in the CDSL version of pwg?

Yes, there is an issue where we had a prolonged discussion about two years back, for changing the Devanagari accent notations in PWG/pwk.

**Comment by fxru** [comment]
Sorry, I don't have any helpful insight. I would agree with @Andhrabharati that replicating the Devanagari of the print faithfully is important. Else, I would suggest to pick one system and stick with it. I think pwg transferred all examples to the Rigveda style accent marking even if the examples come from other Vedas, so the result should be mostly good. However, there were a few exceptions – from the Śatapatha Brāhmaṇa, if I remember correctly – where the correspondence between Devanagari and Latin would probably be different.

**Comment by Yuzki** [comment]
I am the CDSD user who provided this feedback.

I think that it is desirable to use Latin transliteration, although, at the current stage, it seems acceptable to use underlining and vertical lines above as accent marks (rather than acute and grave). As @fxru said, there are different accent markings, such as in Śatapatha Brāhmaṇa, and the marking rule of Śatapatha would involve interpretations of the text. Therefore it would be best to stick with the same accent marking system as Devanagari when transliterating into Latin script.

**Comment by Andhrabharati** [comment]
Vertical line in Roman transliteration has a special meaning, so it cannot be used to represent the 'regular' svarita (as in Devanagari).

**Comment by fxru** [comment]
Could someone help me with understanding the issue better?

1. Does the issue involve only examples or also headwords.
1. Is the problem general to all examples or an issue with specific examples?
1. Is the issue following a pattern (certain source text? Certain prosodic constellations?) that we can identify (semi-) automatically?
1. What is the pattern?
1. Is there disagreement on the nature of the issue or “only” on the solution?

I might not have understood the issue at hand completely.

**Comment by Andhrabharati** [comment]
1 & 2. As I understand, it is in the examples and mostly a limited number.

3 & 4. Need to look at the examples little closer, to get any idea/conclusion.

5. The issue is quite valid; and the solution, though not impossible, is "involved" to implement [and probably not worth Jim's time; he is the ONLY person that has to do all the 'work' here].

**Comment by Yuzki** [comment]
1. It involves only examples.
2. I think it is common if the examples are from the Rigveda and other Vedas with the same accent markings.
3. & 4. Following the Rigveda convention, transliteration into Latin script should be uniquely represented.
5. We share a common understanding of the nature of the issue.


By the way, I looked at the [Max Müller](https://archive.org/details/b30094306_0006/page/524/mode/2up) version and found न्न underlined.
<img width="309" alt="スクリーンショット 2023-06-12 211314" src="https://github.com/sanskrit-lexicon/csl-orig/assets/42864533/ccac2c06-85d2-4ac8-8225-0f35c77a01f5">
Even though I am unfamiliar with Devanagari, I have noticed some discrepancies in the example sentence of the printed version of PWG. However, it is possible that these discrepancies are actually errors. If that is the case, it seems that transliteration could be done automatically based on the rule that a syllable with an acute accent continues from the underline to the next underline or the vertical line above.

**Comment by Andhrabharati** [comment]
Yes, this one is having an error in PWG.
[I had presumed earlier that the links avl. at CDSL for RV are with the text of Max Müller's edition; did not check the actual M. M. edition or other standard works, like from the Vedic Samsodhan Mandal.]

**Does this mean that we can not rely on the digitised texts, as much as on the original prints' scans?**
`----------------------------`
Here is the snippet from the VSM, Pune--
![image](https://github.com/sanskrit-lexicon/csl-orig/assets/75209130/0c3294f6-0850-4162-8227-808201ea2953)

**Comment by Yuzki** [comment]
The accent error in PWG is a flaw on this part, and at least for RV, the reliability of digitized texts should be high.

> But wonder, how and why Aufrecht rendered it without the anudAtta accent!??

Aufrecht represents high-pitched accents with acute and falling accents with grave. He does not indicate low-pitched (=anudātta) syllables, as they are unmarked in the system.

**Comment by gasyoun** [comment]
> Vedaweb has resorted to the ISO transliteration (which has a wide and full coverage of all Indic scripts), and not to IAST (which has some limitations)

Yes, but uses combined characters for example for ṛ and it becomes a mess in searching or copypasting.

**Comment by Yuzki** [comment]
I have tried writing code (I can only write in Python, so please forgive me) to determine where to place acute and grave marks in the SLP1 string. It worked well in the example where the specific PWG error is corrected: `{#a\sU\yanna\Bya^cAkaSa\M tasmA^ aspfhaya\M puna^H#}`.

```
def add_accent_mark():
    # (modified) example sentence
    characters = list('{#a\sU\yanna\Bya^cAkaSa\M tasmA^ aspfhaya\M puna^H#}')

    # determine the position of the vowel
    vowel_position = [False for _ in range(len(characters))]
    vowel_set =  ('a', 'A', 'i', 'I', 'u', 'U', 'f', 'F', 'x', 'X', 'e', 'E', 'o', 'O')
    for i in range(len(characters)):
        if characters[i] in vowel_set:
            vowel_position[i] = True

    # vowel indexes
    vowel_index_list = [i for i in range(len(vowel_position)) if vowel_position[i] is True]

    # set acute and grave marks (They are provisional symbols and don't follow any specific rule!)
    GRAVE_MARK = '>'
    ACUTE_MARK = '<'

    # set anudAtta flag
    flag_anudatta = False
    flag_first_anudatta = False

    for i in range(len(vowel_index_list) - 1):
        vowel_index = vowel_index_list[i]

        # ascertain the character next to the following vowel
        if characters[vowel_index + 1] == '\\':
            flag_first_anudatta = True

            # if the vowel is between underlining and vertical stroke, it has a falling accent
            if characters[vowel_index_list[i + 1] + 1] == '^':
                characters[vowel_index_list[i + 1]] += GRAVE_MARK
                continue
                
            flag_anudatta = True
            continue
        
        else:
            # until the first underlining appears, vowels without any accent mark have a high-pitch accent
            if flag_first_anudatta is True:
                pass
            else:
                characters[vowel_index] += ACUTE_MARK
        

        if flag_anudatta is True:

            # if the character next to the vowel is backslash (= underlining), continue to the next step
            if characters[vowel_index + 1] == '\\':
                continue
            
            # if the character next to the vowel is caret (= vertical stroke), set flag FALSE
            elif characters[vowel_index + 1] == '^':
                flag_anudatta = False
            
            # if the character next to the vowel is neither backslash nor caret (= if it is a consonant character), the vowel has a high-pitch accent
            else:
                characters[vowel_index] += ACUTE_MARK


    return ''.join(characters)
    # OUTPUT: {#a<sU\ya<nna\Bya>^cAkaSa\M ta<smA^ aspfhaya\M pu<na^H#}
```

**Comment by gasyoun** [comment]
> I can only write in Python

@Yuzki would you be willing to help Cologne and write more Python code? We need to add additional features, but none of us knows enough Python, other than Jim who already is busy for next 10 years.

**Comment by Yuzki** [comment]
> @Yuzki would you be willing to help Cologne and write more Python code? We need to add additional features, but none of us knows enough Python, other than Jim who already is busy for next 10 years.

Of course, if I can be of any help, I will help to the best of my ability within the limits of my available time.

**Comment by Yuzki** [comment]
Regarding accent marking in formats other than the Rigveda (RV) style: 

-  Śatapatha Brāhmaṇa (ŚB): No accent marks in the printed version, although the original sentence in the Weber edition has accent marks.
- Maitrāyaṇī Saṃhitā (MS): PWG does not have examples from MS because the Schroeder edition did not exist at that time. 

Conclusion: It is unnecessary to consider accent marking styles of the ŚB or MS in PWG, and it is sufficient to focus on the RV style, which is currently under discussion.

**Comment by drdhaval2785** [comment]
I have not gone through this issue in detail.
The issue has been flagged quite often.
https://github.com/sanskrit-lexicon/csl-ldev/issues/7 and https://github.com/sanskrit-lexicon/csl-ldev/issues/9 may also provide some clue.
